nck/writers/objectstorage_writer.py

```diff
@@ -0,0 +1,78 @@
+# GNU Lesser General Public License v3.0 only
+# Copyright (C) 2020 Artefact
+# licence-information@artefact.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import os
+
+from nck.config import logger
+from nck.writers import Writer
+from nck.utils.retry import retry
+
+
+class ObjectStorageWriter(Writer):
+    def __init__(self, bucket_name, prefix=None, file_name=None, platform=None, **kwargs):
+        self._bucket_name = bucket_name
+        self._prefix = prefix if prefix else ""
+        self._file_name = file_name
+        self._platform = platform
+        self._bucket = self._get_bucket_if_exist()
+
+    @retry
+    def write(self, stream):
+        logger.info(f"Start writing file to {self._platform} ...")
+        print(self._file_name)
+        self._set_valid_file_name(stream.name)
+        print(self._file_name)
+        final_name = os.path.join(self._prefix, self._file_name)
+        self._create_blob(final_name, stream)
+        logger.info(f"Wrote {final_name} file to {self._bucket_name} on  {self._platform} ...")
+        uri = self._get_uri(final_name)
+        logger.info(f"file can be found at {uri}")
+
+    def _get_bucket_if_exist(self):
+        client = self._create_client()
+        bucket = self._create_bucket(client)
+        list_buckets_names = [bucket.name for bucket in self._list_buckets(client)]
+        try:
+            assert self._bucket_name in list_buckets_names
+        except AssertionError as err:
+            logger.exception(f"{self._bucket_name} bucket does not exist.")
+            logger.exception(f"available buckets are {list_buckets_names}")
+            raise err
```

Comment:
Nice use of `logger.exception()`, I was not aware of this method that is indeed perfect for this use case. However, when you raise the exception, it will log something, won't it? I'm afraid the information you log before will be lost because there will be something coming afterwards. Have you tested this behaviour?

Reply:
It log what i want but the trace of the exception is print afterward, so your right, the information is quite lost in too much information. Any idea on how correct that?

Reply:
would this be a better way of doing it ?
```suggestion
            logger.exception(f"{self._bucket_name} bucket does not exist. available buckets are {list_buckets_names}")
            sys.exit(1)
```

Reply:
Hm maybe you can change the message of the exception. Something like:

```suggestion
            err.message = f"{self._bucket_name} bucket does not exist. available buckets are {list_buckets_names}"
            raise err
```

Didn't test it, so tell me if it doesn't work.

Reply:
Hm this seems like it is an old-fashioned way. Maybe you can use [with_traceback](https://docs.python.org/3/library/exceptions.html#BaseException.with_traceback) to take advantage of Python 3